### PR TITLE
dtoh: Generate fieldwise constructors for aggregates

### PR DESCRIPTION
**Pull request by MoonlightSentinel**
This replicates the D behaviour on the C++ side and is required for generated ctors / default arguments which invoke a single argument ctor.

---

~~Depends on #11937~~

### CODE REVIEW

**Review by thewilsonator — APPROVED** [review]

**Comment by kinke** [comment]
Thanks. - C++11 inline initialization of the fields (+ default ctor `<name>() = default;`) would be nice. ;)

File: test/compilable/dtoh_StructDeclaration.d [hunk 1]
```diff
@@ -70,6 +76,12 @@ struct S2
         c()
     {
     }
+    S2(int32_t a, int32_t b = 0, int64_t c = 0LL, S d = S(0, 0, 0LL, nullptr)) :
+        a(a),
+        b(b),
+        c(c),
+        d(d)
+        {}
```
Comment (kinke) [inline test/compilable/dtoh_StructDeclaration.d:84]: This makes `S2 s{123};` ambiguous due to existing `S2(int32_t)` ctor in line 71.

Comment (MoonlightSentinel) [reply test/compilable/dtoh_StructDeclaration.d]: Fixed.

**Review by RazvanN7 — APPROVED** [review]
@MoonlightSentinel This seems to be blocked by a conflict in the changelog. Can you please fix it so that we can get this important addition in?

**Comment by MoonlightSentinel** [comment]
@RazvanN7 Done

**Comment by RazvanN7** [comment]
Auto-merge toggled on. Thanks!

**Comment by MoonlightSentinel** [comment]
> Thanks. - C++11 inline initialization of the fields (+ default ctor `<name>() = default;`) would be nice. ;)

Indeed. I've opted for the current solution because the code was already there (except a few minor changes) and it's another step towards making `frontend.h` compilable.

**Pull request merged**